corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py

```diff
@@ -364,19 +365,23 @@ def update_log(action, doc_ids):
                     obj.save()
         update_log("Created", [obj._migration_couch_id for obj in creates])
         update_log("Updated", [obj._migration_couch_id for obj in updates])
-        update_log("Ignored", [doc["_id"] for doc in ignored])
-        self.ignored_count += len(ignored)
+        update_log("Ignored", ignored_ids)
+        self.ignored_count += len(ignored_ids)
+        return set(ignored_ids)
 
-    def _verify_docs(self, docs, logfile, verify_only):
+    def _verify_docs(self, docs, logfile, verify_only, ignored_ids):
         sql_class = self.sql_class()
         couch_id_name = getattr(sql_class, '_migration_couch_id_name', 'couch_id')
         couch_ids = [doc["_id"] for doc in docs]
         objs = sql_class.objects.filter(**{couch_id_name + "__in": couch_ids})
         objs_by_couch_id = {obj._migration_couch_id: obj for obj in objs}
         diff_count = self.diff_count
         for doc in docs:
-            if verify_only and self.should_ignore(doc):
-                self.ignored_count += 1
+            if verify_only:
+                if self.should_ignore(doc):
+                    self.ignored_count += 1
+                    continue
+            elif doc["_id"] in ignored_ids:
```

Comment:
Would a doc ever show up in `ignored_ids` and also return true for `should_ignore`? I'm wondering if this elif condition should come before the if condition above.

Reply:
If the doc is in `ignored_ids` then `should_ignore` has already been called for that doc and returned true. Passing `ignored_ids` from `_migrate_docs` to `_verify_docs` is a performance optimization so `should_ignore` does not need to be called twice for every document.

Reply:
> I'm wondering if this elif condition should come before the if condition above.

It should not because when `verify_only == True` then `ignored_ids` will be `None` because `_migrate_docs` is not called in that scenario. It is important that this `elif` is not executed if `verify_only` is true.